pennylane/tape/tape.py

```diff
@@ -349,6 +349,18 @@ def __exit__(self, exception_type, exception_value, traceback):
         finally:
             QuantumTape._lock.release()
 
+    def __iter__(self):
+        self._iter_idx = 0
+        return self
+
+    def __next__(self):
```

Comment:
Nice and clean solution in my opinion! It is great to make this object as an iterable, it is more user friendly. This type of addition should make it `Tape` an easier object to use!